File: Bronze1-1/06-22-pm/unit06/water-container/water.java
/*
 * Water container
 */
import java.util.*;
public class water {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int N = in.nextInt(); //# of vertical lines
		int[] H = new int[N];
		for(int j=0; j<N; j++)
			H[j] = in.nextInt();
		in.close();
		
		int maxW = 0;
		for(int i=0; i<N; i++) {
			for(int j=i+1; j<N; j++) {
				//the water height is determined by the min of H[i] and H[j]
				int y = Math.min(H[i], H[j]);
				maxW = Math.max(maxW, y*(j-i));
			}
		}
		System.out.println(maxW);
	}
}

File: Bronze1-1/06-22-pm/unit06/water-container/water.cpp
/*
 * Water container
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N; cin >> N; //# of vertical lines
	int H[N];
	for(int j=0; j<N; j++)
		cin >> H[j];
	
	int maxW = 0;
	for(int i=0; i<N; i++) {
		for(int j=i+1; j<N; j++) {
			//the water height is determined by the min of H[i] and H[j]
			int y = min(H[i], H[j]);
			maxW = max(maxW, y*(j-i));
		}
	}
	cout << maxW << '\n';
}

